src/ufoLib2/typing.py

```diff
@@ -1,4 +1,35 @@
-from typing import Union
+import os
+import sys
+from typing import TypeVar, Union
+
+from fontTools.pens.basePen import AbstractPen
+from fontTools.pens.pointPen import AbstractPointPen
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+
+T = TypeVar("T")
+"""Generic variable for mypy for trivial generic function signatures."""
 
 Number = Union[int, float]
 """Used when integers and floats are interchangable according to the specification."""
+
+PathLike = Union[str, bytes, os.PathLike]
+"""Represents a path in various possible forms."""
+
+
+class Drawable(Protocol):
```

Comment:
sometimes an object only supports draw or drawPoints, but not both. And one doesn't usually use both at the same time. Requiring that a function that uses the draw method receives an objects that must also support the drawPoints method seems too strict to me. It would make sense to split this in two, but I don't know how to name the two classes.

Reply:
We only use `draw()` in ufoLib2, so I cut `drawPoints()` out.